packages/studio/src/components/ComponentNode.tsx

```diff
@@ -30,25 +30,52 @@ interface ComponentNodeProps {
  */
 export default function ComponentNode(props: ComponentNodeProps): JSX.Element {
   const { componentState, depth, isOpen, onToggle, hasChild } = props;
-  const [activeComponentUUID, setActiveComponentUUID] = useStudioStore(
-    (store) => {
-      return [
-        store.pages.activeComponentUUID,
-        store.pages.setActiveComponentUUID,
-      ];
-    }
-  );
+  const [
+    activeComponentUUID,
+    setActiveComponentUUID,
+    selectedComponentUUIDs,
+    addSelectedComponentUUID,
+    clearSelectedComponents,
+    addShiftSelectedComponentUUIDs,
+  ] = useStudioStore((store) => {
+    return [
+      store.pages.activeComponentUUID,
+      store.pages.setActiveComponentUUID,
+      store.pages.selectedComponentUUIDs,
+      store.pages.addSelectedComponentUUID,
+      store.pages.clearSelectedComponents,
+      store.pages.addShiftSelectedComponentUUIDs,
+    ];
+  });
 
   const isActiveComponent = activeComponentUUID === componentState.uuid;
+  const isSelectedComponent = selectedComponentUUIDs.has(componentState.uuid);
 
   const vectorClassName = classNames("cursor-pointer", {
     "rotate-90": isOpen,
     invisible: !hasChild,
   });
 
-  const handleClick = useCallback(() => {
-    setActiveComponentUUID(componentState.uuid);
-  }, [componentState.uuid, setActiveComponentUUID]);
+  const handleClick = useCallback(
+    (event: MouseEvent<HTMLDivElement>) => {
+      if (event.shiftKey && activeComponentUUID) {
+        clearSelectedComponents();
```

Comment:
Since this is in the `if` and `else` I think you can bring this outside the check

Reply:
updated!